### PR TITLE
Fix/snowbox improvements

### PR DESCRIPTION
**Pull request by warm-coolguy**
## Summary

- fix current mapConfiguration
- add ReverseGeocoder
- improve .html display (both snowbox and GitHub page)

Plugins and type package have been slightly altered to fit the usage.

## Instructions for local reproduction and review

Run the snowbox and roughly test all the features.

### CODE REVIEW

File: packages/clients/snowbox/src/index.html [hunk 1]
```diff
@@ -1,7 +1,7 @@
 <!DOCTYPE html>
 <html>
   <head>
-    <title>POLAR@2 Snowbox ❄️📦</title>
+    <title>POLAR@2 Snowbox ❄️📦 Dev</title>
```
Comment (dopenguin) [inline packages/clients/snowbox/src/index.html:4]:
```suggestion
    <title>POLAR Snowbox ❄️📦 Dev</title>
```

File: packages/clients/snowbox/src/index.html [hunk 2]
```diff
@@ -67,29 +67,13 @@
...
-    <p>
-      Since POLAR is, in its core, a library to code with, each package comes
-      with its own documentation. For an overview of what exists, a compiled
-      documentation is also available: See
-      <a id="docs-link" target="_blank">here</a>.
-    </p>
-    <p>
-      A POLAR-generated documentation will always contain exactly the plugins in
-      use.
+    <h1>POLAR@2 Snowbox ❄️📦 Dev</h1>
```
Comment (dopenguin) [inline packages/clients/snowbox/src/index.html:70]:
```suggestion
    <h1>POLAR Snowbox ❄️📦 Dev</h1>
```

File: packages/clients/snowbox/src/mapConfiguration.ts [hunk 3]
```diff
@@ -66,21 +67,19 @@ export const mapConfiguration = {
     layerAttributions: [
       {
         id: basemapId,
-        title: 'Basemap © basemap.de / BKG <MONTH> <YEAR>',
+        title: 'snowbox.layers.basemap',
```
Comment (dopenguin) [inline packages/clients/snowbox/src/mapConfiguration.ts:70]:
```suggestion
        title: 'snowbox.attributions.basemap',
```

File: packages/clients/snowbox/src/mapConfiguration.ts [hunk 4]
```diff
@@ -66,21 +67,19 @@ export const mapConfiguration = {
...
       },
       {
         id: basemapGreyId,
-        title: 'Basemap Grau © basemap.de / BKG <MONTH> <YEAR>',
+        title: 'snowbox.layers.basemapGrey',
```
Comment (dopenguin) [inline packages/clients/snowbox/src/mapConfiguration.ts:74]:
```suggestion
        title: 'snowbox.attributions.basemapGrey',
```

File: packages/clients/snowbox/src/mapConfiguration.ts [hunk 5]
```diff
@@ -66,21 +67,19 @@ export const mapConfiguration = {
...
       },
       {
         id: uBahn,
-        title:
-          'Strecken U-Bahn © Freie und Hansestadt Hamburg, Behörde für Wirtschaft, Verkehr und Innovation',
+        title: 'snowbox.layers.underground',
```
Comment (dopenguin) [inline packages/clients/snowbox/src/mapConfiguration.ts:78]:
```suggestion
        title: 'snowbox.attributions.underground',
```

File: packages/clients/snowbox/src/mapConfiguration.ts [hunk 6]
```diff
@@ -66,21 +67,19 @@ export const mapConfiguration = {
...
       },
       {
         id: sBahn,
-        title:
-          'Strecken S-Bahn © Freie und Hansestadt Hamburg, Behörde für Wirtschaft, Verkehr und Innovation',
+        title: 'snowbox.layers.rapid',
```
Comment (dopenguin) [inline packages/clients/snowbox/src/mapConfiguration.ts:82]:
```suggestion
        title: 'snowbox.attributions.rapid',
```

File: packages/plugins/GeoLocation/src/store/getters.ts [hunk 7]
```diff
@@ -28,6 +28,9 @@ const getters: PolarGetterTree<GeoLocationState, GeoLocationGetters> = {
     }
     return false
   },
+  renderType: (_, __, ___, rootGetters): string => {
+    return rootGetters.configuration?.geoLocation?.renderType || 'independent'
+  },
```
Comment (dopenguin) [inline packages/plugins/GeoLocation/src/store/getters.ts:33]: Return Type should be `RenderType` from the types-package

File: packages/plugins/Legend/CHANGELOG.md [hunk 8]
```diff
@@ -1,5 +1,9 @@
 # CHANGELOG
 
+## unpublished
+
+- Fix: Locale string layer names are now translated.
```
Comment (dopenguin) [inline packages/plugins/Legend/CHANGELOG.md:5]:
```suggestion
- Feature: Locale string layer names are now translated.
```

File: packages/types/custom/CHANGELOG.md [hunk 9]
```diff
@@ -1,5 +1,10 @@
 # CHANGELOG
 
+## unpublished
+
+- Feature: Add parameter `renderType` to `GeoLocationConfiguration`.
+- Feature: Add type for `LegendConfiguration` and add it to `MapConfig`.
```
Comment (dopenguin) [inline packages/types/custom/CHANGELOG.md:6]: Can't seem to see the change in `packages/types/custom/core.ts` 😅

**Review by dopenguin — CHANGES_REQUESTED** [review]
🏓 @warm-coolguy

File: packages/plugins/GeoLocation/src/store/getters.ts (hunk 7 above)
Comment (warm-coolguy) [reply packages/plugins/GeoLocation/src/store/getters.ts]: Yep. https://github.com/Dataport/polar/pull/21/commits/80762b2257660d8b43a71cc1d74b032fea970fcb

File: packages/types/custom/CHANGELOG.md (hunk 9 above)
Comment (warm-coolguy) [reply packages/types/custom/CHANGELOG.md]: Oh right, I've removed that again. Good catch. https://github.com/Dataport/polar/pull/21/commits/396325232ee40f29bcef566059879e6b0a57e788

**Comment by warm-coolguy** [comment]
> I encountered some problems on mobile (Android):
> 
>     * The GeoLocation-button is always visible and always disabled. I didn't get a notification regarding locating my device.
>       Screenshot_2023-11-16-10-33-48-22_40deb401b9ffe8e1df2f1cc5ba480b12
> 
>     * Holding the phone vertically, I can't see to the see the whole map, even if I try to zoom out of the website.
>       Screenshot_2023-11-16-10-33-57-21_40deb401b9ffe8e1df2f1cc5ba480b12

1. This is actually the correct behaviour. The browser rejects GeoLocation requests on remote HTTP requests. Annoying for testing, but not an issue, and it will work in the deployed version on our GitHub page.
2. I can't reproduce this. But while we're here, I made the client full-width on the page, it was somewhat narrow on my mobile. Have you tried using another browser on your mobile? I don't think this problem is related to the current MR.

**Comment by warm-coolguy** [comment]
> 🏓 @warm-coolguy

🏓 @dopenguin

**Review by dopenguin — APPROVED** [review]
Changing the width of map not only fixed the issue for me but also we now have a practical example for the overlay regarding the zoom.

🚀 @warm-coolguy

**Pull request merged**